tests/serialization/particle_handler_02.cc

```diff
@@ -107,20 +107,12 @@ test()
             << " is in cell " << particle->get_surrounding_cell(tr)
             << std::endl;
 
-  // TODO: Move this into the Particle handler class. Unfortunately, there are
-  // some interactions with the SolutionTransfer class that prevent us from
-  // doing this at the moment. When doing this, check that transferring a
-  // solution and particles during the same refinement is possible (in
-  // particular that the order of serialization/deserialization is preserved).
-  tr.signals.pre_distributed_save.connect(std::bind(
-    &Particles::ParticleHandler<dim,
-                                spacedim>::register_store_callback_function,
-    &particle_handler));
-
   // save data to archive
   std::ostringstream oss;
   {
     boost::archive::text_oarchive oa(oss, boost::archive::no_header);
+
+    particle_handler.prepare_for_serialization();
```

Comment:
I would call this function directly before the `save` function in line 131. As far as I know, this function only connects the callback function to a signal in `save`, right?